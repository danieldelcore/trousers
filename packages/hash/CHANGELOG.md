# @trousers/hash

## 3.0.0

### Major Changes

-   e6a3323: # Version 3 🎉

    This is Trousers' biggest update yet!

    Including:

    -   Support for the CSS prop 👩‍🎤
    -   Mono-repo setup and decomposition 📦
    -   Object notation CSS 🍞
    -   Theme CSS var package 🎨
    -   insertRule 🏹
    -   Smaller bundles 🐭
    -   Custom style collectors 🕺
    -   Trousers package with all exports for backwards compatibility

    ## Removed

    -   withStyles HOC (please use css prop for class components)

    ### Developer experience

    -   Adds [storybook-addon-performance](https://github.com/atlassian-labs/storybook-addon-performance)
    -   Adds [react-stable-ref](https://github.com/danieldelcore/react-stable-ref) to test for re-renders
    -   Adds [atlassian/changesets](https://github.com/atlassian/changesets)
    -   Adds [manypkg](https://github.com/Thinkmill/manypkg)

-   c9c8187: **Trousers v3!**

    -   CSS prop 👩‍🎤
    -   Mono-repo setup and decomposition 📦
    -   Object notation CSS 🍞
    -   Theme CSS var package 🎨
    -   prod mode support for `insertRule` 🏹
    -   Smaller bundles 🐭
    -   Support for custom style collectors 🕺
    -   Trousers package with all exports for backwards compatibility

    **Dev**

    -   Adds [storybook-addon-performance](https://github.com/atlassian-labs/storybook-addon-performance)
    -   Adds [react-stable-ref](https://github.com/danieldelcore/react-stable-ref) to test for re-renders
    -   Adds [atlassian/changesets](https://github.com/atlassian/changesets)
