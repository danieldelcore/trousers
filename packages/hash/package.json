{
    "name": "@trousers/hash",
    "version": "3.0.0",
    "main": "dist/index.js",
    "module": "dist/index.js",
    "types": "dist/index.d.ts",
    "license": "MIT",
    "dependencies": {},
    "devDependencies": {},
    "repository": "https://github.com/danieldelcore/trousers/tree/master/packages/hash"
}
