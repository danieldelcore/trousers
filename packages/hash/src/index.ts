export { default as generateHash } from './hash-generator';
