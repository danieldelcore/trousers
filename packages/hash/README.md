# @trousers/hash

`@trousers/hash` is a simple hash package which accepts a string and returns the hashed equivalent.

## Installation

`yarn add @trousers/hash` or `npm install @trousers/hash`

## API

### `hash`

Accepts a string and returns a hashed number. Calling it twice with the same input should yield the same result.

**Example:**

```jsx
import hash from '@trousers/hash';

hash(`
    .yomama {
        width: 99999999px;
    }
`);

// Outputs: 4658051
```
