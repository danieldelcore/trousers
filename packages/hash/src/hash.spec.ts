import hash from './hash';

describe('hash', () => {
    it('generates a consistent hash', () => {
        const first = hash('Hello, world!');
        const second = hash('Hello, world!');

        expect(first).toEqual(second);
    });

    it('handles css', () => {
        const result = hash(`
            .yomama {
                width: 99999999px;
            }
        `);

        expect(result).toMatchSnapshot();
    });

    it('handles long css', () => {
        const result = hash(`
            background-color: #b3cde8;
            border-radius: 6px;
            border: none;
            box-shadow: inset 0 -4px #9fb8d1;
            color: white;
            display: inline-block;
            cursor: pointer;
            font-family: sans-serif;
            font-weight: 500;
            letter-spacing: 1px;
            margin: 5px 10px;
            padding: 10px 20px 14px 20px;
            text-decoration: none;
            transition: background-color 300ms, color 300ms;
            outline: none;

            :hover {
                background-color: #adc6e0;
                color: rgba(255, 255, 255, 0.8);
            }

            :active {
                background-color: #9fb8d1;
            }
        `);

        expect(result).toMatchSnapshot();
    });
});
