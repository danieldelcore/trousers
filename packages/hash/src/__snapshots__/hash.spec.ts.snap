// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash handles css 1`] = `4658051`;

exports[`hash handles long css 1`] = `276237355`;
